### PR TITLE
Upgrade Solana deps to 1.16.5

### PR DESCRIPTION
**Pull request by Aursen**
* Upgrade Solana deps to 1.16.5
* Upgrade Rust version to 1.69.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ yarn start # start local dev server
 
 ```sh
 rustc --version
-# rustc 1.65.0 (897e37553 2022-11-02)
+# rustc 1.69.0 (84c898d65 2023-04-16)
```
Comment (acheroncrypto) [inline README.md:49]: I see that the Solana crates bumped the MSRV to `1.68.0`, let's update this to latest stable `rustc 1.71.0 (8ede3aae2 2023-07-12)` while we are at it.

File: wasm/rust-toolchain.toml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "stable-2022-11-03"
+channel = "stable-2023-04-20"
```
Comment (acheroncrypto) [inline wasm/rust-toolchain.toml:2]: 👍

File: README.md (hunk 1 above)
Comment (Aursen) [reply README.md:49]: In the Solana repo they use the 1.69 rust version but yeah anyways let's use the latest

**Review by acheroncrypto — APPROVED** [review]
Thank you.

**Pull request merged**